packages/datadog-instrumentations/src/prisma.js

```diff
@@ -66,23 +66,21 @@ class TracingHelper {
 addHook({ name: '@prisma/client', versions: ['>=6.1.0'] }, (prisma, version) => {
   const tracingHelper = new TracingHelper()
 
-  /*
-    * This is a custom PrismaClient that extends the original PrismaClient
-    * This allows us to grab additional information from the PrismaClient such as DB connection strings
-  */
-  class PrismaClient extends prisma.PrismaClient {
-    constructor (...args) {
-      super(...args)
-
-      const datasources = this._engine?.config.inlineDatasources?.db.url?.value
+  // we need to patch the prototype to get db config since this works for ESM and CJS alike.
+  const originalRequest = prisma.PrismaClient.prototype._request
+  prisma.PrismaClient.prototype._request = function () {
```

Comment:
This will run on each request while formerly, we only needed to check this during the construction time of the instance.
I am not sure why replacing the constructor works for CJS but not for ESM. Do you know why that's the case?

If we only ever need the first instance, we could just replace the method again afterwards to the original one, while I wonder if there's another thing that we could hook into that also contains the necessary information.

Another issue I see is if a customer potentially has more than a single Prisma instance. I am not sure if that's even possible. I guess Prisma does not allow that?

Reply:
So prisma does allow multiple instances, which means we probably should change the code to allow 1 db config block per instantiated client. We could also, once we save the db config, just reset the request method on the instance (not the prototype) to the original request. 

Not sure why the constructor approach wasn't working, though the code for Prisma is bundled so its not easy to look at the src code.

Reply:
For now, the comment explains why we're doing what we're doing here, so that's fine.